examples/Protonect.cpp

```diff
@@ -48,6 +48,27 @@ void sigint_handler(int s)
   protonect_shutdown = true;
 }
 
+bool protonect_paused = false;
+libfreenect2::Freenect2Device *devtopause;
+
+//Doing non-trivial things in signal handler is bad. If you want to pause,
+//do it in another thread.
+//Though libusb operations are generally thread safe, I cannot guarantee
+//everything above is thread safe when calling start()/stop() while
+//waitForNewFrame().
+void sigusr1_handler(int s)
+{
+  if (devtopause == 0)
+    return;
+/// [pause]
+  if (protonect_paused)
```

Comment:
I looked at it more closely. `wait_for` is actually quite non-trivial to get right.

One other alternative is to catch the signal in the main loop and pause right there until one more signal, and then continue the main loop. But this still needs a bunch of `#ifdef` to use platform dependent timing functions or signal functions.